pkg/kubernetes/scripts/policy.js

```diff
@@ -174,6 +177,66 @@
                 return result;
             }
 
+            var sarCache = { };
+
+            function subjectAccessReview(project, user, verb, resource) {
+                var key = (project || "") + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+                // cached already?
+                if (key in sarCache)
+                    return sarCache[key];
+
+                // pre-fill cache to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                loader.watch("Projects").then(function() {
```

Comment:
Hm, is that what `watch()` does? It's documented as "The returned promise will be resolved when an initial set of objects have been loaded for the watch, or rejected if the watch has failed." which makes it sound like an one-time trigger. What would be a better API then? `loader.listen()` and explicitly `.cancel()` it  right away?